e2e/testcases/preserve_fields_test.go

```diff
@@ -198,12 +195,12 @@ aggregationRule:
 	nt.WaitForRepoSyncs()
 
 	// Ensure the aggregate rule is actually aggregated.
-	duration, err := nomostest.Retry(20*time.Second, func() error {
-		return nt.Validate(aggregateRoleName, "", &rbacv1.ClusterRole{}, clusterRoleHasRules([]rbacv1.PolicyRule{
-			nsViewer.Rules[0], rbacViewer.Rules[0],
-		}))
-	})
-	nt.T.Logf("took %v to wait for aggregate ClusterRole", duration)
+	err := nomostest.WatchObject(nt, kinds.ClusterRole(), aggregateRoleName, "",
```

Comment:
should this specify the same 20 second timeout?

Reply:
I can add a 30s timeout. It seemed like the default would be ok after WaitForRepoSyncs, but the default is pretty long.

Reply:
Added.